### ISSUE TITLE
[BUG] EventHubBufferedProducerClient: Flushing the client results in ObjectDisposedException on subsequent enqueueing of events

### ISSUE THREAD

**Issue by LouisSkyline** [Azure/azure-sdk-for-net#32871]
### Library name and version

Azure.Messaging.EventHubs 5.7.5

### Describe the bug

After a FlushAsync has been called on the client, all subsequent enqueue calls fail on the client with a System.ObjectDisposedException.

### Expected behavior

Either it should be possible to enqueue events after a flush without exceptions being thrown or the documentation should be updated indicating that no events can be enqueued after calling the flush.

### Actual behavior

Calling EnqueueEventAsync after calling a FlushAsync throws the following exception:

> System.ObjectDisposedException: The CancellationTokenSource has been disposed.
   at System.Threading.CancellationTokenSource.ThrowObjectDisposedException()
   at Azure.Messaging.EventHubs.Producer.EventHubBufferedProducerClient.<StartPublishingAsync>d__90.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Messaging.EventHubs.Producer.EventHubBufferedProducerClient.<EnqueueEventsAsync>d__75.MoveNext()

### Reproduction Steps

```
        var client = new EventHubBufferedProducerClient("<connection string>", new EventHubBufferedProducerClientOptions()
        {
            EnableIdempotentRetries = true,
            MaximumWaitTime = TimeSpan.FromSeconds(1),
            MaximumEventBufferLengthPerPartition = 100,
            RetryOptions = new EventHubsRetryOptions()
            {
                Mode = EventHubsRetryMode.Exponential,
                MaximumDelay = TimeSpan.FromSeconds(30),
            }
        });
        client.SendEventBatchFailedAsync += HandleFailed; 
        client.SendEventBatchSucceededAsync += HandleSuccess; 

        
        await client.EnqueueEventAsync(new EventData("This is my first event"));

        await client.FlushAsync();

        await client.EnqueueEventAsync(new EventData("This is my second event")); //Throws ObjectDisposedException
```

### Environment

.NET SDK:

- Version:   7.0.100
- Commit:    e12b7af219

Runtime Environment:

- OS Name:     Windows
- OS Version:  10.0.22621
- OS Platform: Windows
- RID:         win10-x64

IDE:

- Visual Studio 2022
- Console Application
- Target Framework: .NET 6

**Comment by jsquire** [comment]
Hi @LouisSkyline.  Thank you for reaching out and we regret that you're experiencing difficulties.  The intent is for you to be able to continue to use the buffered producer after flushing; enqueueing new events should work.   It looks like we don't have test coverage over this specific scenario; I'll take a look at reproducing and getting a fix for this.

**Comment by LouisSkyline** [comment]
Any update on this? When can we expect the next release?

**Comment by jsquire** [comment]
Hi @LouisSkyline.   Apologies for the lack of communication; this got held up longer than expected in validation and missed the February release window.    The release is now expected the second week of March and is being tracked by #34422.  We'll be sure to communicate updates there.

**Comment by jsquire** [comment]
Hi @LouisSkyline.   Thank you for your patience.  This has been released as part of v5.8.0.
